static/src/javascripts/projects/common/utils/Queue.js

```diff
@@ -0,0 +1,42 @@
+define([], function (
```

Comment:
helps with encapsulation as well. in the steady page example we're working on a queue with defined operations instead of messing about with an array declared in the module